tests/unit/amo/reducers/testReviews.js

```diff
@@ -891,5 +892,81 @@ describe(__filename, () => {
       const state = addReviewToState({ state: prevState, review });
       expect(state.byUserId[userId]).not.toBeDefined();
     });
+
+    it('resets ratingSummary when adding a new rating', () => {
+      const addonId = 44231;
+      let state;
+
+      state = reviewsReducer(
+        state,
+        setRatingSummary({
+          addonId,
+          summary: { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 },
+        }),
+      );
+
+      state = addReviewToState({
+        state,
+        review: createInternalReview({
+          ...fakeReview,
+          addon: {
+            ...fakeReview.addon,
+            id: addonId,
+          },
+        }),
+      });
+
+      expect(state.ratingSummary[addonId]).not.toBeDefined();
+    });
+  });
+
+  describe('setRatingSummary', () => {
+    it('stores a rating summary', () => {
+      const addonId = 432;
+      const summary = {
+        1: 64,
+        2: 122,
+        3: 456,
+        4: 1243,
+        5: 922,
+      };
+
+      const state = reviewsReducer(
+        undefined,
+        setRatingSummary({
+          addonId,
+          summary,
+        }),
+      );
+
+      expect(state.ratingSummary[addonId]).toEqual(summary);
+    });
+
+    it('preserves existing rating summaries', () => {
```

Comment:
does each version have it's own summary too? No, right?

Reply:
actually it might but not sure if there is a use case for this..?

Reply:
Hmm, I need to think about that for a minute.

Reply:
It is correct as is.

See https://github.com/mozilla/addons-frontend/issues/4529#issuecomment-400836661:

"we shouldn't be filtering ratings by version."

Good question. I always get confused when I think of this because the old site was way more complicated.

Reply:
> does each version have it's own summary too?

btw, the answer to this is technically yes because you can make a request to the API endpoint with `version=N` and you'll only get ratings submitted for that version.